tedge_config/src/models/flag.rs

```diff
@@ -0,0 +1,85 @@
+use std::convert::{TryFrom, TryInto};
+
+/// Represents a boolean type.
+///
+/// We need this newtype in order to implement `TryFrom<String>` and `TryInto<String>`.
+/// The config_key macro uses query_string() and update_string().
+/// Therefore, boolean needs to be converted from/to String.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq)]
+#[serde(transparent)]
+pub struct Flag(pub bool);
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to parse flag: {input}. Supported values are: true, false")]
+pub struct InvalidFlag {
+    input: String,
+}
+
+impl TryFrom<String> for Flag {
+    type Error = InvalidFlag;
+
+    fn try_from(input: String) -> Result<Self, Self::Error> {
+        match input.as_str() {
+            "true" => Ok(Flag(true)),
+            "false" => Ok(Flag(false)),
+            _ => Err(InvalidFlag { input }),
+        }
+    }
+}
+
+impl TryInto<String> for Flag {
+    type Error = std::convert::Infallible;
+
+    fn try_into(self) -> Result<String, Self::Error> {
+        match self {
+            Flag(true) => Ok(String::from("true")),
+            Flag(false) => Ok(String::from("false")),
+        }
+    }
+}
```

Comment:
The `TryFrom<T>` implicitly provides `TryInto<T>`  so I think the `TryInto` implementation is redundant.
```suggestion
impl TryFrom<String> for Flag {
    type Error = InvalidFlag;

    fn try_from(input: String) -> Result<Self, Self::Error> {
        match input.as_str() {
            "true" => Ok(Flag(true)),
            "false" => Ok(Flag(false)),
            _ => Err(InvalidFlag { input }),
        }
    }
}
```

Reply:
I think you say that because of: 
```shell
 the trait bound `std::string::String: From<Flag>` is not satisfied
  --> tedge/src/cli/config/config_key.rs:32:48
   |
32 |                 config.update_string($setting, value)
   |                                                ^^^^^ the trait `From<Flag>` is not implemented for `std::string::String`
...
```

If that is so you just need to implement `From<String>`:
```rust
impl From<Flag> for String {
    fn from(val: Flag) -> Self {
        val.0.to_string()
    }
}
```
This will work because your Flag type is jsut a wrapper for a `bool` which implements `FromStr` and therefore can be used to convert.

So the code probably could look like this:
```suggestion
impl TryFrom<String> for Flag {
    type Error = InvalidFlag;

    fn try_from(input: String) -> Result<Self, Self::Error> {
        match input.as_str() {
            "true" => Ok(Flag(true)),
            "false" => Ok(Flag(false)),
            _ => Err(InvalidFlag { input }),
        }
    }
}

impl From<Flag> for bool {
    fn from(val: Flag) -> Self {
        val.0
    }
}

impl From<Flag> for String {
    fn from(val: Flag) -> Self {
        val.0.to_string()
    }
}
```

Reply:
Oh yes, You are right. Thanks!